pom.xml

```diff
@@ -7,21 +7,37 @@
     <groupId>pl.javastart</groupId>
     <artifactId>euler2</artifactId>
     <version>1.0</version>
+    <dependencies>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <version>RELEASE</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
 
     <build>
         <plugins>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>2.22.1</version>
+                <version>2.19.1</version>
                 <dependencies>
                     <dependency>
                         <groupId>org.junit.platform</groupId>
                         <artifactId>junit-platform-surefire-provider</artifactId>
-                        <version>1.2.0</version>
+                        <version>1.3.1</version>
                     </dependency>
                 </dependencies>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>7</source>
+                    <target>7</target>
+                </configuration>
+            </plugin>
```

Comment:
To powoduje zmianę wersji Javy dla tego projektu na Java 7, co jest w sumie jakimś starociem i nie ma prawa działać. Lepiej podać coś nowszego, np ostatni LTE, czyli wersja 17:
```java
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <configuration>
                    <source>17</source>
                    <target>17</target>
                </configuration>
            </plugin>
```

Natomiast domyślnie są tutaj używane parametry `maven.compiler.source` oraz `maven.compiler.target`, czyli coś takiego:

```xml
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <configuration>
                    <source>{maven.compiler.source}</source>
                    <target>{maven.compiler.target}</target>
                </configuration>
            </plugin>
```

więc można im przypisać odpowiednie wartości, wtedy nie trzeba wchodzić w szczegóły konfiguracji tego pluginu. Najczęściej propertiesy ustawia się zaraz po nazwie grupy i projektu:

```xml
    <groupId>pl.javastart</groupId>
    <artifactId>euler2</artifactId>
    <version>1.0</version>

    <properties>
        <maven.compiler.source>17</maven.compiler.source>
        <maven.compiler.target>17</maven.compiler.target>
    </properties>
```